pinot-common/src/main/java/org/apache/pinot/common/exception/QueryException.java

```diff
@@ -125,8 +126,17 @@ public static void setMaxLinesOfStackTrace(int maxLinesOfStackTrace) {
   }
 
   public static ProcessingException getException(ProcessingException processingException, Exception exception) {
+    return getException(processingException, getTruncatedStackTrace(exception));
+  }
+
+  public static ProcessingException getException(ProcessingException processingException, String errorMessage) {
     String errorType = processingException.getMessage();
     ProcessingException copiedProcessingException = processingException.deepCopy();
+    copiedProcessingException.setMessage(errorType + ":\n" + errorMessage);
+    return copiedProcessingException;
+  }
+
+  public static String getTruncatedStackTrace(Exception exception) {
```

Comment:
Not sure if we should expose this method

Reply:
Why not? One example of its usefulness is in this PR as we need to truncate the stack trace and return it in broker response.

Reply:
To send stack trace, we should use `getException(ProcessingException processingException, Exception exception)` instead of calling this method to construct the error message and use `getException(ProcessingException processingException, String errorMessage)`

Reply:
`getException` method takes `ProcessingException` as its argument while we need to construct a `QueryProcesssingException` to pass it in broker response. That's why i did that refactoring.

Reply:
Sorry didn't realize they are different classes.
(Optional) You might also want to keep the `errorType` information in the error message? If so, you can construct a `ProcessingException` first, then use it to construct the `QueryProcessingException`: `new QueryProcessingException(exception.getErrorCode(), exception.getMessage())`

Reply:
`errorType`, which is basically `exception.getMessage()`, is always available in the first line of the stack trace. Since we keep the stack trace, we don't need to separately keep it.

Reply:
The `errorType` I referred to here is `BrokerResponseSendError` which you put in the `QueryException.class` so that users don't need to look up the int error code (425) to understand the type of the error. The error message of the Exception will be:
```
  BrokerResponseSendError:
  {stacktrace}
```

I already approved the change. These comments are just suggestions. You can merge it

Reply:
I don't have the write access to merge this PR. How can I get the write access?

Reply:
@sajjad-moradi Merged the PR. I think you need to be a committer in order to merge PR. @mayankshriv @mcvsubbu Do we need to start a vote in order to make Sajjad a committer?